public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -62,6 +64,30 @@ export class PanelQueryRunner {
     return this.panelId;
   }
 
+  /**
+   * Get the last result -- optionally skip the transformation
+   */
+  //  TODO: add tests
+  getCurrentData(transform = true, depth?: number): PanelData {
+    const v = this.state.validateStreamsAndGetPanelData();
+    if (transform && this.transformations && this.transformations.length) {
+      // When retrieving the data for the transformations editor, we want the input data to be transformed using onle preceeding transformers
+      // This will be a performance bottleneck for large data sets as transformations will be executed multiple times for differnt editors
+      // We need to implement memoization for the transformers to optimize it
+      const transformationsToApply = depth !== undefined ? this.transformations.slice(0, depth) : this.transformations;
```

Comment:
Do we really need to expose the depth here?  I would think the editor could execute them directly when it needs to, keeping the kinda weird/complex use out of `panelQueryRunner`

Reply:
Yes, you are right. I have removed that now and the pre-transform data is calculated in the editors itself